addons/wowl/static/src/components/navbar/navbar.ts

```diff
@@ -16,14 +16,58 @@ export class NavBar extends Component<{}, OdooEnv> {
   actionManager = useService("action_manager");
   menuRepo = useService("menus");
 
+  private _debouncedAdapt = utils.debounce(this._adapt.bind(this), 250);
+
   constructor(...args: any[]) {
     super(...args);
     hooks.onMounted(() => {
-      this.env.bus.on("MENUS:APP-CHANGED", this, () => this.render());
+      this.env.bus.on("MENUS:APP-CHANGED", this, () => {
+        this._adapt();
+      });
     });
+
+    hooks.useExternalListener(window, "resize", this._debouncedAdapt);
   }
 
   systrayItems = this._getSystrayItems();
+  private _overflowingAppSections: string[] = [];
+  private _showEmptyMoreMenu = false;
+
+  private async _adapt() {
+    // ------- Initialize -------
+    // Render all (needed to retrieve render widths)
+    this._overflowingAppSections = [];
+    this._showEmptyMoreMenu = true;
+    await this.render();
+
+    // Check initialization
+    const sectionsMenu = this.el!.querySelector<HTMLElement>(".o_menu_sections")!;
+
+    // ------- Check overflowing sections -------
+    const sectionsElements = this.el!.querySelectorAll<HTMLElement>(
+      ".o_menu_sections > *:not(.o_menu_sections_more)"
+    );
+    const sectionsAvailableWidth = sectionsMenu.offsetWidth;
+    const sectionsContentWidth = [...sectionsElements].reduce((sum, el) => sum + el.offsetWidth, 0);
+
+    if (sectionsAvailableWidth < sectionsContentWidth) {
+      // Determine which sections are overflowing
+      const moreDropdown = this.el!.querySelector<HTMLElement>(".o_menu_sections_more");
+      let width = moreDropdown!.offsetWidth;
+      for (const el of sectionsElements) {
+        if (sectionsAvailableWidth >= width + el.offsetWidth) {
+          width += el.offsetWidth;
+        } else {
+          const sectionId = el.querySelector("[data-section]")!.getAttribute("data-section")!;
+          this._overflowingAppSections.push(sectionId);
+        }
+      }
```

Comment:
Note: this loop could lead to an unconsistent ordering of the sections items.

![image](https://user-images.githubusercontent.com/1159815/97418397-11dc9180-1909-11eb-98c8-381da4519efe.png)

Reply:
```suggestion
      for (const el of sectionsElements) {
        if (sectionsAvailableWidth < width + el.offsetWidth) {
          const sections = [...sectionsElements];
          const overflowingSections = sections.slice(sections.indexOf(el));
          this._overflowingAppSections = overflowingSections.map(overflowingEl => {
            return overflowingEl.querySelector("[data-section]")!.getAttribute("data-section")!
          });
          break;
        }
        width += el.offsetWidth;
      }
```